<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/iLogos/logo_novafct.png" width="200">

# Departamento de Engenharia Mecânica e Industrial
## Mecânica dos Sólidos II

## Tensões e deformações na flexão pura

### Problema 1

Determine a relação entre o peso de três vigas com o mesmo comprimento, para as quais o mesmo momento fletor origina a mesma tensão normal máxima. As três secções transversais estão indicadas na figura e têm a forma, respetivamente, de um círculo, um quadrado e um retângulo de proporção $h = 2b$.

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/Notebooks/Au01/P1/MSII_Au01_P1.png" width="800"/>

#### Nomenclatura
* $W$ : módulo resistente de uma viga [mm $^3$] 
* $I_z$ : momento de inércia de área ou segundo momento de área [mm $^4$]
* $P$ : peso da viga [kg]
* $\rho$ : massa volúmica [kg/mm³]
* $V$ : volume [mm³]

### Resolução

Uma vez que as vigas têm o mesmo comprimento ($L$) e a mesma massa volúmica ($\rho$), a sua diferença de peso ($P=\rho V = \rho A L$) será apenas determinada pela área de secção de cada perfil. A área do perfil circular é definida pelo raio $R$, a área do perfil quadrado por $l$ e finalmente a secção retangular pela dimensão $l$ (uma vez que existe a relação $h = 2b$).

- Questões de análise:
    - Dos três perfis qual deles apresenta maior resistência à flexão?
    - Qual dos três perfis necessita de uma menor área de secção (menor peso) para resistir ao momento fletor aplicado, para um comprimento fixo da viga?
    - Que quantidades físicas são relevantes na caraterização da secção transversal de um elemento submetido a flexão? Qual o significado destas grandezas?

**Esquema da resolução:**

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSII/Notebooks/Au01/P1/MSII_Au01_P1_AnsoGraphic.png"
 width="1000"/>

### Resolução 1

Para cada secção, comece-se por calcular as grandezas: momento de inércia de área ($I_z$), maior distância ao centroide da secção ($c$), área da secção ($A$), momento resistente ($W$).

#### Secção circular

- Momento de inércia de área:

\begin{equation*}
I_z^\bigcirc = \int_A y²~\mathrm{d}A = \int_{0}^{R} r² \left[\int_{0}^{2\pi} \sin² (\alpha)~r \mathrm{d}\alpha \right] \mathrm{d}r =
\frac{\pi R⁴}{4} \quad\mathrm{(mm^4)}
\end{equation*}

In [17]:
import sympy as sy

t, r, R = sy.symbols('t r R')

Ii = sy.integrate(sy.sin(t)**2*r,(t,0,2*sy.pi))
Icirc = sy.integrate(Ii*r**2,(r,0,R))
print(f' I.circ = {Icirc}')

 I.circ = pi*R**4/4
ola


- Distância máxima em relação ao Eixo Neutro (E.N.):

\begin{equation*}
c^\bigcirc \equiv y_\mathrm{max} = R \quad\mathrm{(mm)}
\end{equation*}

- Módulo de resistência à flexão:

\begin{equation*}
\left(W = \frac{I_z}{c} \right)^\bigcirc = \frac{\pi R⁴}{4}\left(\frac{1}{R}\right) = \frac{\pi R^3}{4}
\quad\mathrm{(mm^3)}
\end{equation*}

In [2]:
ycirc = R
print(f' y.circ = {ycirc}')
Wcirc = Icirc/ycirc
print(f' W.circ = {Wcirc}')

 y.circ = R
 W.circ = pi*R**3/4


- Condição de dimensionamento:

\begin{equation*}
\left(\sigma_\mathrm{adm} \ge \frac{M_f}{W} \right)^\bigcirc \ge \frac{4 M_f}{\pi R³}
\quad\mathrm{(MPa)}
\end{equation*}

resolvendo para $R$,

\begin{equation*}
R \ge \left( \frac{4 M_f}{\pi \sigma_\mathrm{adm}} \right)^{1/3}
\quad\mathrm{(mm)}
\end{equation*}

In [3]:
Sadm, M = sy.symbols('Sadm M')

eq1 = Sadm - M/Wcirc
sol1 = sy.solve(eq1,{R})

Rcirc = sy.simplify(sol1[0])
print(f' R.circ = {Rcirc}')

 R.circ = 2**(2/3)*(M/Sadm)**(1/3)/pi**(1/3)


- Peso da viga:

\begin{equation*}
P^\bigcirc = \rho V = \rho AL = \rho~\underbrace{\pi R^2}_{A^\bigcirc}~L = \rho \pi^{1/3} 4^{2/3} \left( \frac{M_f}{\sigma_\mathrm{adm}} \right)^{2/3} L
\quad\mathrm{(Kg)}
\end{equation*}

onde,
- $\rho$ massa volúmica (kg/mm³)
- $V$ volume (mm³)

In [4]:
Acirc = sy.pi*Rcirc**2
print(f' Acirc = {Acirc}')

L, rho = sy.symbols('L rho')
Pcirc = rho*Acirc*L
print(f' Pcirc = {Pcirc}')

 Acirc = 2*2**(1/3)*pi**(1/3)*(M/Sadm)**(2/3)
 Pcirc = 2*2**(1/3)*pi**(1/3)*L*rho*(M/Sadm)**(2/3)


### Secção quadrada

- Momento de inércia de área:

\begin{equation*}
I_z^\square = \int_A y²~\mathrm{d}A = \int_{-b/2}^{b/2} \left(\int_{-h/2}^{h/2} y^2~\mathrm{d}y \right) \mathrm{d}x =
\frac{l⁴}{12} \quad\wedge\quad l=b =h~\mathrm{(mm^4)}
\end{equation*}

In [5]:
l, x, y = sy.symbols('l x y')

Ii = sy.integrate(y**2,(y,-l/2,l/2))
Iquad = sy.integrate(Ii,(x,-l/2,l/2))
print(f' I.quad = {Iquad}')

 I.quad = l**4/12


- Distância máxima em relação ao eixo neutro:

\begin{equation*}
c^\square = \frac{l}{2} \quad\mathrm{(mm)}
\end{equation*}

- Módulo de resistência à flexão:

\begin{equation*}
\left(W = \frac{I_z}{c} \right)^\square
= \frac{l⁴}{12} \left(\frac{2}{l}\right)
= \frac{l^3}{6}
\quad\mathrm{(mm^3)}
\end{equation*}

In [6]:
yquad = l/2
print(f' y.quad = {yquad}')
Wquad = Iquad/yquad
print(f' W.quad = {Wquad}')

 y.quad = l/2
 W.quad = l**3/6


- Condição de dimensionamento:

\begin{equation*}
\left(\sigma_\mathrm{adm} \ge \frac{M_f}{W} \right)^\square
\ge \frac{6 M_f}{l³}
\quad\mathrm{(MPa)}
\end{equation*}

resolvendo para $l$,

\begin{equation*}
l \ge \left( \frac{6 M_f}{\sigma_\mathrm{adm}} \right)^{1/3}
\quad\mathrm{(mm)}
\end{equation*}

In [7]:
eq2 = Sadm - M/Wquad
sol2 = sy.solve(eq2,{l})

lquad = sy.simplify(sol2[0])
print(f' l.circ = {lquad}')

 l.circ = 6**(1/3)*(M/Sadm)**(1/3)


- Peso da viga:

\begin{equation*}
P^\square = \rho V = \rho AL
= \rho~\underbrace{l^2}_{A^\square}~L
= \rho 6^{2/3} \left( \frac{M_f}{\sigma_\mathrm{adm}} \right)^{2/3} L
\quad\mathrm{(Kg)}
\end{equation*}

In [8]:
Aquad = lquad**2
print(f' Aquad = {Aquad}')

Pquad = rho*Aquad*L
print(f' Pquad = {Pquad}')

 Aquad = 6**(2/3)*(M/Sadm)**(2/3)
 Pquad = 6**(2/3)*L*rho*(M/Sadm)**(2/3)


### Secção retangular

- Momento de inércia de área:

\begin{equation*}
I_z^\sqsubset = \int_A y²~\mathrm{d}A = \int_{-b/2}^{b/2} \left(\int_{-h/2}^{h/2} y^2~\mathrm{d}y \right) \mathrm{d}x =
\frac{bh³}{12}\Big|_{h=2b} = \frac{2b⁴}{3} \quad\wedge\quad h=2b~\mathrm{(mm^4)}
\end{equation*}

In [9]:
b = sy.symbols('b')
h = 2*b

Ii = sy.integrate(y**2,(y,-h/2,h/2))
Iretang = sy.integrate(Ii,(x,-b/2,b/2))
print(f' I.retang = {Iretang}')

 I.retang = 2*b**4/3


- Distância máxima em relação ao eixo neutro:

\begin{equation*}
c^\sqsubset = \frac{h}{2} = b \quad\mathrm{(mm)}
\end{equation*}

- Módulo de resistência à flexão:

\begin{equation*}
\left(W = \frac{I_z}{c} \right)^\sqsubset
= \frac{2b⁴}{3} \left(\frac{1}{b}\right)
= \frac{2b^3}{3}
\quad\mathrm{(mm^3)}
\end{equation*}

In [10]:
yretang = h/2
print(f' y.retang = {yretang}')
Wretang = Iretang/yretang
print(f' W.retang = {Wretang}')

 y.retang = b
 W.retang = 2*b**3/3


- Condição de dimensionamento:

\begin{equation*}
\left(\sigma_\mathrm{adm} \ge \frac{M_f}{W} \right)^\sqsubset
\ge \frac{3 M_f}{2b³}
\quad\mathrm{(MPa)}
\end{equation*}

resolvendo para $b$,

\begin{equation*}
b \ge \left( \frac{3\, M_f}{2\, \sigma_\mathrm{adm}} \right)^{1/3}
\quad\mathrm{(mm)}
\end{equation*}

In [11]:
eq3 = Sadm - M/Wretang
sol3 = sy.solve(eq3,{b})

bretang = sy.simplify(sol3[0])
print(f' b.retang = {bretang}')

 b.retang = 2**(2/3)*3**(1/3)*(M/Sadm)**(1/3)/2


- Peso da viga:

\begin{equation*}
P^\sqsubset = \rho V = \rho AL
= \rho~\underbrace{2b^2}_{A^\sqsubset}~L
= \rho 2^{1/3} 3^{2/3} \left( \frac{M_f}{\sigma_\mathrm{adm}} \right)^{2/3} L
\quad\mathrm{(Kg)}
\end{equation*}

In [12]:
Aretang = 2*bretang**2
print(f' Aretang = {Aretang}')

Pretang = rho*Aretang*L
print(f' Pretang = {Pretang}')

 Aretang = 2**(1/3)*3**(2/3)*(M/Sadm)**(2/3)
 Pretang = 2**(1/3)*3**(2/3)*L*rho*(M/Sadm)**(2/3)


As três expressões para os peso próprio, $P^\bigcirc$, $P^\square$
$P^\sqsubset$, têm em comum o termo:

\begin{equation*}
k = \rho \left( \frac{M_f}{\sigma_\mathrm{adm}} \right)^{2/3} L
\end{equation*}

Pelo que resulta,

| Círculo   |      Quadrado      |  Retângulo |
|----------|:-------------:|------:|
| $P^\bigcirc = \pi^{1/3}4^{2/3} k $ |  $P^\square = 6^{2/3} k$ | $P^\sqsubset = 2^{1/3} 3^{2/3} k$ |

In [13]:
# k = L*rho*(M/Sadm)**(2/3)
PcircN = float(Pcirc.subs({L:1,rho:1,M:1,Sadm:1}))
PquadN = float(Pquad.subs({L:1,rho:1,M:1,Sadm:1}))
PretangN = float(Pretang.subs({L:1,rho:1,M:1,Sadm:1}))
print(f' P.circN  = {PcircN:.5f} k')
print(f' P.quadN  = {PquadN:.5f} k')
print(f'P.retangN = {PretangN:.5f} k')

 P.circN  = 3.69054 k
 P.quadN  = 3.30193 k
P.retangN = 2.62074 k


Calculando rácios de peso entre as secções:

In [14]:
frac1 = PcircN/PquadN
print(f'P.circ/P.quad   = {frac1:.5f} ({frac1*100:.1f} %)')
frac2 = PretangN/PquadN
print(f'P.retang/P.quad = {frac2:.5f} ({frac2*100:.1f} %)')

frac3 = PcircN/PretangN
print(f'\nP.circ/P.retang = {frac3:.5f} ({frac3*100:.1f} %)')
frac4 = PquadN/PretangN
print(f'P.quad/P.retang = {frac4:.5f} ({frac4*100:.1f} %)')

frac5 = PquadN/PcircN
print(f'\nP.quad/P.circ   = {frac5:.5f} ({frac5*100:.1f} %)')
frac6 = PretangN/PcircN
print(f'P.retang/P.circ = {frac6:.5f} ({frac6*100:.1f} %)')

P.circ/P.quad   = 1.11769 (111.8 %)
P.retang/P.quad = 0.79370 (79.4 %)

P.circ/P.retang = 1.40820 (140.8 %)
P.quad/P.retang = 1.25992 (126.0 %)

P.quad/P.circ   = 0.89470 (89.5 %)
P.retang/P.circ = 0.71012 (71.0 %)


## Resolução 2

- A relação entre o peso das 3 vigas é igual à relação entre as áreas da secção transversal, para o mesmo comprimento das vigas ($P = \rho A L $, com $\rho$ e $L$ constantes)

- Para que o mesmo momento fletor, $M$, produza a mesma tensão máxima é necessário que as 3 secções tenham o mesmo módulo resistente, $W$:

\begin{equation*}
| \sigma_\mathrm{max} | = \frac{M_f}{W}
\rightarrow W = \frac{I_z}{c}
\end{equation*}

#### **Secção circular** $\bigcirc$

\begin{equation*}
I_z = \frac{\pi R⁴}{4};~c = R
\quad\therefore\quad W_\bigcirc = \frac{I_z}{c} = \frac{\pi R³}{4}
\end{equation*}

#### **Secção quadrada** $\square$

\begin{equation*}
I_z = \frac{l⁴}{12};~c = \frac{l}{2}
\quad\therefore\quad W_\square = \frac{I_z}{c} = \frac{l⁴}{12}
\left( \frac{2}{l} \right) = \frac{l^3}{6}
\end{equation*}

#### **Secção retangular** $\sqsubset$

\begin{equation*}
I_z = \frac{bh³}{12} = \frac{8b⁴}{12};~c = \frac{h}{2}=b
\quad\therefore\quad W_\sqsubset = \frac{I_z}{c} = \frac{8b⁴}{12}
\left( \frac{1}{b} \right) = \frac{2b^3}{3}
\end{equation*}

#### **Condição de igualdade dos módulos resistentes. Part I**

\begin{equation*}
{W_\bigcirc} = {W_\square}
\quad\therefore\quad
\frac{\pi R³}{4} = \frac{l^3}{6}
\end{equation*}

resolvendo,

\begin{equation*}
R = \left(\frac{2 l^3}{3 \pi }\right)^{1/3}
\end{equation*}

#### Relação entre secções

\begin{equation*}
\mathrm{rac.1} = \frac{{A_\bigcirc}}{{A_\square}} =
\frac{\pi R²}{l²} = \frac{\pi}{l}\left(\frac{2}{3 \pi }\right)^{2/3}
\end{equation*}

Desenvolvendo a análise por unidade de cumprimento: $l=1$:

\begin{equation*}
\mathrm{rac.1} = \pi\left(\frac{2}{3 \pi }\right)^{2/3}
= 1.118
\end{equation*}

In [15]:
import numpy as np

r1 = np.pi*(2/3/np.pi)**(2/3)
print(f'r1 = {r1:.4f} ({r1*100:.1f} %)')

r1 = 1.1177 (111.8 %)


#### **Condição de igualdade dos módulos resistentes. Part II**

\begin{equation*}
{W_\square} = {W_\sqsubset}
\quad\therefore\quad
\frac{l^3}{6} = \frac{2b^3}{3}
\end{equation*}

resolvendo para $b$,

\begin{equation*}
b = \left(\frac{l^3}{4}\right)^{1/3}
\end{equation*}

#### Relação entre secções

\begin{equation*}
\mathrm{rac.2} = \frac{{A_\sqsubset}}{{A_\square}} =
\frac{2b²}{l²} = \frac{2}{l²}\left(\frac{l^3}{4}\right)^{2/3} = 2\left(\frac{1}{4}\right)^{2/3} = 0.794
\end{equation*}

In [16]:
r2 = 2*(1/4)**(2/3)
print(f'r2 = {r2:.4f} ({r2*100:.1f} %)')

r2 = 0.7937 (79.4 %)


---

Copyright (c) DEMI - FCT NOVA

Interactive computing by <a href="https://jupyter.org/" target="_blank"> <span
style="color:#333399"> Jupyter Notebook </span> </a> &nbsp;|&nbsp;Coded by <a href = "mailto: jmc.xavier@fct.unl.pt">José Xavier</a>

Licensed under  <a href="http://creativecommons.org/licenses/by-sa/4.0/"
target="_blank"> <span style="color:#333399;font-size: 20px"> CC BY-SA 4.0  </span></a>